project-base/storefront/components/Layout/Header/AutocompleteSearch/AutocompleteSearch.tsx

```diff
@@ -1,142 +1,117 @@
+                onBlur={handleBlur}
+            >
+                <SearchInput
+                    className="w-full border-2 border-white max-vl:border-primaryLight"
+                    label={t("Type what you're looking for")}
+                    onEnterPressCallback={() =>
+                        router.push({
+                            pathname: searchUrl,
+                            query: { q: searchQueryValue },
+                        })
+                    }
+                    value={searchQueryValue}
+                    onChange={(e) => setSearchQueryValue(e.currentTarget.value)}
+                    isLoading={isFetchingSearchData}
+                />
+
+                {searchQueryValue.length > 0 && (
+                    <div
+                        className="absolute right-11 top-1/2 flex -translate-y-1/2 cursor-pointer items-center justify-center rounded-full bg-greyLighter p-2"
+                        onClick={clearSearchQuery}
+                    >
+                        <CloseIcon />
+                    </div>
+                )}
+
                 <div
                     className={twJoin(
-                        'transition lg:absolute lg:left-0 lg:top-0 lg:z-[1] lg:max-h-[50px] lg:w-full',
-                        hasAutocompleteSearchFocus && 'lg:max-h-auto',
+                        'absolute left-0 -bottom-3 z-aboveOverlay flex w-full origin-top translate-y-full scale-y-90 flex-col gap-6 bg-creamWhite p-5 px-7 pb-6 shadow-md transition-all lg:rounded',
+                        isSearchResultsPopupVisible
+                            ? 'pointer-events-auto scale-y-100 opacity-100'
+                            : 'pointer-events-none opacity-0',
                     )}
-                    ref={autocompleteSearchInRef}
                 >
-                    <div
-                        className={twJoin(
-                            'relative flex w-full transition-all lg:focus-within:w-[576px]',
-                            hasAutocompleteSearchFocus && 'z-[1021] lg:w-[576px]',
-                        )}
-                    >
-                        <SearchInput
-                            className={twJoin(
-                                'border-2',
-                                hasAutocompleteSearchFocus
-                                    ? 'max-vl:w-full max-vl:!border-primaryLight'
-                                    : 'border-white',
-                            )}
-                            label={t("Type what you're looking for")}
-                            dataTestId={TEST_IDENTIFIER}
-                            onEnterPressCallback={onAutocompleteSearchHandler}
-                            value={autocompleteSearchQueryValue}
-                            onChange={onChangeAutocompleteSearchQueryValueHandler}
-                            isLoading={areSearchResultsLoading}
+                    {isSearchResultsPopupVisible && (
```

Comment:
If I click on the autocomplete popup (outside of the links). It disappears.
[Screencast from 30.8.2023 09:16:16.webm](https://github.com/shopsys/shopsys/assets/55181354/7949dcb6-89e8-413a-9046-6850b6c6e80d)

Reply:
Ah, this is something which I've missed in the implemented logic. It's refactored now.